lib/widgets/recent_dm_conversations.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+
+import '../model/narrow.dart';
+import '../model/recent_dm_conversations.dart';
+import 'content.dart';
+import 'icons.dart';
+import 'message_list.dart';
+import 'page.dart';
+import 'store.dart';
+import 'text.dart';
+
+class RecentDmConversationsPage extends StatefulWidget {
+  const RecentDmConversationsPage({super.key});
+
+  static Route<void> buildRoute({required BuildContext context}) {
+    return MaterialAccountPageRoute(context: context,
+      builder: (context) => const RecentDmConversationsPage());
+  }
+
+  @override
+  State<RecentDmConversationsPage> createState() => _RecentDmConversationsPageState();
+}
+
+class _RecentDmConversationsPageState extends State<RecentDmConversationsPage> with PerAccountStoreAwareStateMixin<RecentDmConversationsPage> {
+  RecentDmConversationsView? model;
+
+  @override
+  void onNewStore() {
+    model?.removeListener(_modelChanged);
+    model = PerAccountStoreWidget.of(context).recentDmConversationsView
+      ..addListener(_modelChanged);
+  }
+
+  void _modelChanged() {
+    setState(() {
+      // The actual state lives in [model].
+      // This method was called because that just changed.
+    });
+  }
+
+  Widget _buildItem(BuildContext context, DmNarrow narrow) {
```

Comment:
I think it's probably a bit more idiomatic to break this out as its own widget.

There's some discussion of this in the StatelessWidget docs:
https://api.flutter.dev/flutter/widgets/StatelessWidget-class.html
complete with a video "Widgets vs Helper Methods". I think the reasons there why one sometimes really strongly wants separate widgets don't apply here, but it's still an indication of the direction in which to lean.